### PR TITLE
Visualisation Fixes Part II

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description
Improvements for the visualization refactoring. 

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md), [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.
- [x] All code has been profiled where possible.

### CODE REVIEW

File: src/rust/lib/graph-editor/src/component/visualization/foreign/java_script/definition.rs [hunk 1]
```diff
@@ -93,36 +93,46 @@
 //!   area automatically. The `size` parameter contains two fields `width` and `height` expressed in
 //!   pixels.
 
-// FIXME: Make sure the above definition is implemented in the JS API, including the `Visualization`
-//        superclass, its `dom` field and the `setPreprocessor` method (connected to Rust side).
+// TODO: Connect the `setPreprocessor` method on Rust side.
```
Comment (wdanilo) [inline src/rust/lib/graph-editor/src/component/visualization/foreign/java_script/definition.rs:96]: is the setPreprocessor method from JS not connected to the FRP yet? Can we please do it in this PR? This is the last piece to have fully functional specification connected. Moreover, after talking with engine today, it seems that we will need this functionality very, very soon.

File: src/rust/lib/graph-editor/src/component/visualization/foreign/java_script/definition.rs [hunk 2]
```diff
@@ -93,36 +93,46 @@
...
 // === Constants ===
 // =================
 
-const LABEL_FIELD      : &str = "label";
-const INPUT_TYPE_FIELD : &str = "inputType";
+#[allow(missing_docs)]
+pub mod field {
+    pub const LABEL         : &str = "label";
+    pub const INPUT_TYPE    : &str = "inputType";
+    pub const INPUT_FORMAT  : &str = "inputFormat";
```
Comment (wdanilo) [inline src/rust/lib/graph-editor/src/component/visualization/foreign/java_script/definition.rs:127]: alignment

File: src/rust/lib/graph-editor/src/component/visualization/foreign/java_script/instance.rs [hunk 3]
```diff
@@ -129,39 +134,40 @@ impl InstanceModel {
         scene.manage(&self.root_node);
     }
 
+    fn set_size(&self, size:V2) {
+        let size          = Vector2::new(size.x,size.y);
+        let data_json     = JsValue::from_serde(&size).unwrap();
+        let _             = self.try_call1(&self.set_size, &data_json);
+        self.root_node.set_size(size);
```
Comment (wdanilo) [inline src/rust/lib/graph-editor/src/component/visualization/foreign/java_script/instance.rs:141]: alignment

File: src/rust/lib/graph-editor/src/component/visualization/definition.rs [hunk 4]
```diff
@@ -20,22 +20,28 @@ use visualization::java_script;
 #[allow(missing_docs)]
 pub struct Signature {
     #[shrinkwrap(main_field)]
-    pub path       : visualization::Path,
-    pub input_type : EnsoType,
+    pub path         : visualization::Path,
+    pub input_type   : EnsoType,
+    pub input_format : Rc<visualization::data::Format>,
```
[the remark this answers is not in the document]
Comment (MichaelMauderer) [reply src/rust/lib/graph-editor/src/component/visualization/definition.rs:25]: `data::Format` is an enum, so not `CloneRef`able.

**Pull request merged**